### ISSUE TITLE
Tenant not initialized for config files. intervention/imagecache support or workaround

### ISSUE THREAD

**Issue by SebastianBotez** [archtechx/tenancy#475]
#### The BUG
I try to serve images using intervention/imagecache package.
http://image.intervention.io/use/url

This packages allows you to upload images in the original format and then serve them in any resolution or size you want. 
For example http://yourhost.com/{route-name}/{template-name}/{file-name}

The {route-name} needs to be specified in the config file of the package in imagecache.route
As the documentation states, the tenant storage path function is storage_path({folder})
My config settings for config path key is:

`
    'paths' => [
        storage_path('app/uploads/images')
    ],
`

#### Steps to reproduce
1. composer require intervention/imagecache
2. publish the config file
3. in the config file, paths key, add `storage_path('app/uploads/images')` or any other path

#### Expected behavior
The storage_path function used should return `{ProjectName}/storage/{tenant}/app/uploads/images` but it returns `{ProjectName}/storage/app/uploads/images`
This means that the tenant is not identified (maybe?).

#### Your setup
 - Laravel version: 7.15.0
 - stancl/tenancy version: 3

Any workaround or method to fix this behaviour?
Thank you.

**Comment by stancl** [comment]
This is not a bug.

If you need to change the path to be the tenant's storage path, use an event hook to change that config key.

**Comment by SebastianBotez** [comment]
Tried to modify the config file on the fly, tested with some dumps and they were not initialized with custom events. Looks like this class somehow is serving the image before the tenancy events are fired.
Nevermind, I found a workaround for those who will need it:

In config/imagecache.php, replace path key with this closure:

`    'paths' => (function() {
        $tenantDirectories = [];
        foreach (scandir(storage_path()) as $key => $directory) {
            if(strpos($directory, 'tenant') !== false) {
                $tenantDirectories[] = storage_path('{$directory}/app/uploads/images');
            }
        }
        return $tenantDirectories;
    })(),`

A good practice for the code above would be to replace the string 'tenant' with config('tenancy.filesystem.suffix_base') but unfortunatelly it returns null, so I can live with this workaround for now.

Thank you @stancl for your response.
Best regards!
